DA-209: Change Swan to SWAN

judembo opened the pull request.

JeannedArk committed:
DA-15: Improve timeline view and add visualization config in scheme

The width calculation in the timeline is now dynamically calculated depending
on the max width of the neighbour nodes. Subgraphs are displayed in their own
section. Additionally the visualization elements can be configured in the scheme.
The kind if visualization (timeline and graph) can be configured also in the
scheme builder.

JeannedArk committed:
DA-15: Allow visualization configuration in scheme of none, graph and timeline

The scheme allows to define different visualizations of none, graph and timeline.
Furthermore display options were added for e.g. positioning the links in the
timeline mode. Therefore the scheme builder and the scheme preview was adjusted.
The entity models were changed so that every label and linklabel are now identified
by there id and not by there name/ string label, so that configurations to each
label can be applied.
The timeline places the nodes depending on the given configuration.
The visualization options are no exclusive.
Some bugs were fixed according to the displaying of the visualization kinds
schemeUploadModal html and corresponding controller has been refactored.

Commit:
DA-136: Add shortcuts to cycle through links

The shortcuts 'n' and 'm' can now be used to cycle through
the outgoing links of an annotation.

JeannedArk committed:
Merge pull request #230 from annefried/DA-136

DA-136: Add shortcuts to cycle through links

JeannedArk committed:
DA-15: Fixes and displaying annotation options improvements

No (null) elements added when options are undefined in schemeUploadModalController
and fix loading from existing scheme. Fix switching from timeline to graph in the
schemebuilder, creating a linklabel, defining positioning but not adding label.
Entity model adjusted to the new views.

JeannedArk committed:
Merge pull request #227 from annefried/DA-15_2

DA-15: Insert timeline and visualization scheme options

JeannedArk committed:
DA-141: Correct terminology in frontend and backend #1

Rename TargetType to SpanType, startType to startSpanType, endType to endSpanType,
labelId to name, targetType as attribute of TargetType to name.

JeannedArk committed:
Merge pull request #233 from annefried/DA-141

DA-141: Correct terminology in frontend and backend #1

JeannedArk committed:
DA-141: Correct terminology in front- and backend #2, refactor schemeUploadModal

The terminology was further corrected, LinkSet was renamed to LinkType. The
schemeUploadModal and schemeViewModal was refactored. Instead of converting
between two different models (front- and backend) only the backend model
is used now, so no conversion is needed anymore, which caused an overhead
of code. Furthermore they were generally refactored, including creating a
UtilityService. The scheme upload with JSON files was also fixed.

JeannedArk committed:
Merge pull request #236 from annefried/DA-141

DA-141: Correct terminology in front- and backend #2, refactor scheme…

JeannedArk committed:
DA-217: Add support for several tokenization languages

Before English was the default tokenization language. The user can now
choose the prefered tokenization language of a set of several
supported languages.

JeannedArk committed:
DA-238: Optimize timelogging by user request

Optimized timelogging by user request which executed unnecessary database
queries on the user relations.

JeannedArk commented:
I would like to do this in one step, including also renaming the packages, maybe we can rename the project without trouble

JeannedArk committed:
DA-226: Prevent frontend from adding an annotation with the same start, end and span type

It is not possible anymore to add or change an annotation so that there
are two annotations with the same start, end and span type properties.
Additionally some parts of the annotationController and d3Annotation
were refactored.

judembo committed:
Merge pull request #237 from annefried/DA-217

DA-217: Add support for several tokenization languages

judembo committed:
Merge pull request #239 from annefried/DA-238

DA-238: Optimize timelogging by user request

judembo committed:
Merge pull request #240 from annefried/DA-226

DA-226: Prevent frontend from adding an annotation with the same star…

judembo committed:
DA-162: Add schemeUploadModalControllerTest

schemeUploadModalControllerTest.js was added;
several issues in the schemeUploadModalController, mostly related
to loading existing schemes, that were discovered during testing were fixed;
rootControllerTest was fixed;

JeannedArk committed:
DA-162: Update and fix karma npm dependencies

There was an error because angular-clipboard was not included when the
rootControllerTest was executed. Additionally all karma dependencies
were to the latest version updated.

JeannedArk committed:
Merge pull request #241 from annefried/DA-162_4

DA-162: Add schemeUploadModalControllerTest

JeannedArk committed:
DA-238: Improve user requests

When editing a project an optimized user list will be requested without
projects. Adding a user was also optimized and took some time before in
an environment with lots of data.

JeannedArk committed:
DA-244: Refactor frontend and create structure for cross doc linking

A more modularized structure was applied so that it can be
better reused for future tasks including cross doc linking e.g.
the graph and timeline views were separated in extra html files
so that they can be included in other html files.

stgrue committed:
Merge pull request #243 from annefried/DA-238

DA-238: Improve user requests

stgrue committed:
Merge pull request #245 from annefried/DA-244

DA-244: Refactor frontend and create structure for cross doc linking

Commit:
DA-141: Correct terminology and refactor ExportUtil

The names of certain entities have been changed from DiscAnno to Swan.
Additionally, ExportUtil.java has been refactored to remove code duplication
and make it more readable.

JeannedArk committed:
Merge pull request #247 from annefried/DA-141

DA-141: Correct terminology and refactor ExportUtil

Commit:
DA-231: Fix rendering of links to/from off-screen annotations

Previously, links which had an off-screen annotations on one end were not
displayed correctly (either in the wrong direction or not at all). This is
fixed by estimating the positions of unrendered annotations by the position
of their first word so that the line can be drawn correctly.

NOTE: This patch fixes issues #231 and #232, as they are closely related.

JeannedArk committed:
DA-216: last-edited-document column in projects explorer

The column displays the last edited document depending on the users state
and last edited value and links to the last edited document. The column
is only for annotators visible.

JeannedArk committed:
Merge pull request #242 from annefried/DA-216

DA-216: Add go-to-last-edited-document-button in projects explorer

JeannedArk committed:
Merge pull request #248 from annefried/DA-231

DA-231: Fix rendering of links to/from off-screen annotations

judembo committed:
DA-149: Make completion checkbox fixed

A top bar containing project/document information, the completion checkbox
and previous/next buttons was added to the annotation view,
so annotators don't have to scroll up anymore to mark documents as completed.

JeannedArk committed:
Merge pull request #252 from annefried/DA-149_2

DA-149: Make completion checkbox fixed

judembo committed:
DA-209: Change Swan to SWAN

...and update Logo
